src/core/socialprovider.js

```diff
@@ -45,7 +45,8 @@ var XMPPSocialProvider = function(dispatchEvent) {
   // rate limiting).
   this.sendMessagesTimeout = null;
   this.timeOfFirstMessageInBatch = 0;
-  this.messages = [];
+  // buffered outbound messages, arrays of messages keyed by recipient.
```

Comment:
Can you update this now to say that this.messages is a mapping from recipient to message/continuation pair?